libkbfs/disk_block_cache_wrapped.go

```diff
@@ -112,23 +111,48 @@ func (cache *diskBlockCacheWrapped) IsSyncCacheEnabled() bool {
 }
 
 // Get implements the DiskBlockCache interface for diskBlockCacheWrapped.
-func (cache *diskBlockCacheWrapped) Get(ctx context.Context, tlfID tlf.ID,
-	blockID kbfsblock.ID) (
+func (cache *diskBlockCacheWrapped) Get(
+	ctx context.Context, tlfID tlf.ID, blockID kbfsblock.ID,
+	preferredCacheType DiskBlockCacheType) (
 	buf []byte, serverHalf kbfscrypto.BlockCryptKeyServerHalf,
 	prefetchStatus PrefetchStatus, err error) {
 	cache.mtx.RLock()
 	defer cache.mtx.RUnlock()
 	primaryCache := cache.workingSetCache
 	secondaryCache := cache.syncCache
-	if cache.config.IsSyncedTlf(tlfID) && cache.syncCache != nil {
-		primaryCache, secondaryCache = secondaryCache, primaryCache
+	if preferredCacheType == DiskBlockSyncCache {
+		if cache.syncCache != nil {
+			primaryCache, secondaryCache = secondaryCache, primaryCache
+		} else {
+			log := cache.config.MakeLogger("DBC")
+			log.Warning("Sync cache is preferred, but there is no sync cache")
+		}
 	}
 	// Check both caches if the primary cache doesn't have the block.
 	buf, serverHalf, prefetchStatus, err =
-		primaryCache.Get(ctx, tlfID, blockID)
+		primaryCache.Get(ctx, tlfID, blockID, preferredCacheType)
 	if _, isNoSuchBlockError := err.(NoSuchBlockError); isNoSuchBlockError &&
 		secondaryCache != nil {
-		return secondaryCache.Get(ctx, tlfID, blockID)
+		buf, serverHalf, prefetchStatus, err = secondaryCache.Get(
+			ctx, tlfID, blockID, preferredCacheType)
```

Comment:
Agreed, but it is part of the `Get` interface that both `diskBlockCacheWrapped` and `DiskBlockCacheLocal` have to implement.

Reply:
Yeah. I was thinking about that--maybe we need an explicit interface for the wrapped cache, so that we don't have this extraneous parameter in the specific caches.

Reply:
> Yeah. I was thinking about that--maybe we need an explicit interface for the wrapped cache, so that we don't have this extraneous parameter in the specific caches.

True, but that's not really a new thing introduced in this PR.  For example, the unref stuff and tlf-clearning stuff isn't applied to both caches.  It might make sense to do as a separate cleanup, but I don't want to shoehorn it into this PR.